cmd/download.go

```diff
@@ -94,12 +95,22 @@ Download other people's solutions by providing the UUID.
 			req.URL.RawQuery = q.Encode()
 		}
 
-		payload := &downloadPayload{}
-		res, err := client.Do(req, payload)
+		res, err := client.Do(req)
 		if err != nil {
 			return err
 		}
 
+		payload := &downloadPayload{}
```

Comment:
Minor nitpick. Instead of initializing with an a pointer to a downloadPayload I would recommend using `var payload downloadPayload` and then in the call to Decode take the address, as that is the only place we actually need the pointer.

Reply:
Sweet, I'll update that.